src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,128 @@
+import classNames from 'classnames';
+import {
+  FormEvent, useEffect, useRef, useState,
+} from 'react';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo;
+  onDelete: (id: number) => void;
+  updateTodos: (id: number, data: Partial<Todo>) => void;
+  isLoading: boolean;
+  selectedId: number;
+  toggleLoader: boolean;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDelete,
+  updateTodos,
+  isLoading,
+  selectedId,
+  toggleLoader,
+}) => {
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const [doubleClick, setDoubleClick] = useState(false);
+  const [updatedTitle, setUpdatedTitle] = useState(todo.title);
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, [doubleClick]);
+
+  const handleTitleChange = (e: FormEvent) => {
+    e.preventDefault();
+    setDoubleClick(false);
+
+    if (!updatedTitle.trim().length) {
+      onDelete(todo.id);
+    }
+
+    updateTodos(todo.id, { title: updatedTitle });
+  };
+
+  return (
+    <div
+      className={classNames('todo', {
+        completed: todo.completed,
+      })}
+      data-cy="Todo"
+    >
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          defaultChecked={todo.completed}
+          onChange={() => updateTodos(todo.id, { completed: !todo.completed })}
```

Comment:
Pls move the function inside `onChange` handler to a separate const and use as a callback here